src/serial/mod.rs

```diff
@@ -0,0 +1,22 @@
+pub struct Serial {
+    port: usize,
+}
+
+impl Serial {
+    pub fn new(port: usize) -> Self {
+        Self { port }
+    }
+
+    pub fn write(&self, data: &str) {
+        for byte in data.bytes() {
+            self.write_byte(byte);
+        }
+    }
+
+    fn write_byte(&self, byte: u8) {
+        let addr = self.port as *mut u8;
+        unsafe {
+            *addr = byte;
+        }
```

Comment:
I think this part might benefit from using [this function](https://doc.rust-lang.org/std/ptr/fn.write_volatile.html)

This way, the compiler guarantees that the write won't be optimized away, which I'm assuming might be an issue for IO mapped ports

Reply:
Good catch. I will fix this